packages/react/src/Button/Button.tsx

```diff
@@ -6,26 +6,41 @@
 import clsx from 'clsx'
 import { forwardRef } from 'react'
 import type { ButtonHTMLAttributes, ForwardedRef, PropsWithChildren } from 'react'
+import { Icon } from '../Icon'
+import type { IconProps } from '../Icon'
 
 export type ButtonProps = {
+  /** An icon to add to the button. */
+  icon?: IconProps['svg']
+  /** The position of the icon. The default is after the label. */
+  iconPosition?: 'start' | 'only'
   /** The level of prominence. Use a primary button only once per page or section. */
   variant?: 'primary' | 'secondary' | 'tertiary'
 } & PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>
 
 export const Button = forwardRef(
   (
-    { children, className, type, disabled, variant = 'primary', ...restProps }: ButtonProps,
+    { children, className, disabled, icon, iconPosition, type, variant = 'primary', ...restProps }: ButtonProps,
     ref: ForwardedRef<HTMLButtonElement>,
   ) => {
     return (
       <button
         {...restProps}
         ref={ref}
         disabled={disabled}
-        className={clsx('ams-button', `ams-button--${variant}`, className)}
+        className={clsx(
+          'ams-button',
+          `ams-button--${variant}`,
+          iconPosition === 'only' && `ams-button--icon-position-only`,
+          className,
+        )}
         type={type || 'button'}
       >
-        {children}
+        {icon && (iconPosition === 'start' || iconPosition === 'only') && (
+          <Icon svg={icon} size="level-5" square={iconPosition === 'only'} />
```

Comment:
I've changed the `square` prop to prevent the padding from growing too big when a label is shown. The padding around the icon should resemble the padding around the inline text.

Reply:
Gets me thinking… Should it even stay a public prop? We only want the icon container to become square if the icon is in a button and alone, right?

Reply:
https://gemeente-amsterdam.atlassian.net/browse/DES-997